src/common/telemetryHelper.ts

```diff
@@ -203,7 +203,7 @@ export class TelemetryHelper {
         return telemetryProperties;
     }
 
-    public static generate<T>(name: string, codeGeneratingTelemetry: { (telemetry: TelemetryGenerator): Thenable<T> }): Q.Promise<T> {
+    public static generate<T>(name: string, codeGeneratingTelemetry: { (telemetry: TelemetryGenerator): Thenable<T>|T }): Q.Promise<T> {

```

Comment:
Looks like all async code paths have been converted to be promisy now. In that case, do we still need codeGeneratingTelemetry to be of a non-thenable type?